cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -136,6 +129,27 @@ type API struct {
 	BindPort int32
 }
 
+// NodeRegistrationOptions holds fields that relate to registering a new master or node to the cluster, either via "kubeadm init" or "kubeadm join"
+type NodeRegistrationOptions struct {
+
+	// Name is the `.Metadata.Name` field of the Node API object that will be created in this `kubeadm init` or `kubeadm joiń` operation.
+	// This field is also used in the CommonName field of the kubelet's client certificate to the API server.
+	// Defaults to the hostname of the node if not provided.
+	Name string
+
+	// CRISocket is used to retrieve container runtime info. This information will be annotated to the Node API object, for later re-use
+	CRISocket string
+
+	// Taints specifies the taints the Node API object should be registered with. If this field is unset, i.e. nil, in the `kubeadm init` process
+	// it will be defaulted to []v1.Taint{'node-role.kubernetes.io/master=""'}. If you don't want to taint your master node, set this field to an
+	// empty slice, i.e. `taints: {}` in the YAML file. This field is solely used for Node registration.
+	Taints []v1.Taint
+
+	// ExtraArgs passes through extra arguments to the kubelet. The arguments here are passed to the kubelet command line via the environment file
+	// kubeadm writes at runtime for the kubelet to source.
+	ExtraArgs map[string]string
```

Comment:
Given that this is the Kubelet ExtraArgs it's not entirely clear from a developer UX.  

For example: 
NodeRegistrationOptions.ExtraArgs

vs. 
KubeletRegistrationOptions.ExtraArgs or 
NodeRegistrationOptions.KubeletExtraArgs.

Reply:
I'll change this after we've discussed the naming with @fabriziopandini in the SIG meeting